Add navigator

thachnguyendino opened the pull request.

thachnguyendino committed:
Add navigator

thachnguyendino committed:
- Correct UI SignIn.

thachnguyendino committed:
Changes:
- Added api login by facebook.
- Using sharepreferences to store user data for sign out features

thachnguyendino committed:
- Implement Forgot feature

thachnguyendino committed:
Changes:
- Implement fabric crashlytics.

thachnguyendino committed:
- Add tournament.

thachnguyendino committed:
- Add tournament api

ndduy commented:
Resolve conflict please

ndduy commented on SportBook/app/src/main/java/com/dinosys/sportbook/extensions/FragmentExtension.kt:
```diff
@@ -2,5 +2,84 @@ package com.dinosys.sportbook.extensions
+            "Uid" to headerObj.getString("Uid")
+    )
+    return AuthModel(authData, Headers.of(map))
+}
```
Anh không nghĩ mình để nên cái này thành Fragment Extension, nó không trực tiếp liên quan đến View. Để cái này trong LoginManager hợp lý hơn.

ndduy commented on SportBook/app/src/main/java/com/dinosys/sportbook/utils/SPUtil.kt:
```diff
@@ -0,0 +1,23 @@
+        return spPreferences.getString(key, defaultVal)
+    }
+
+}
```
Hạn chế sử dụng từ viết tắt trong tên biến tên class tên object

ndduy commented on SportBook/app/src/main/res/values/strings.xml:
```diff
@@ -1,12 +1,20 @@
     <string name="app_name">Sport Book</string>
     <string name="facebook_app_id">103873786863872</string>
-    <string name="username_text">Username</string>
+    <string name="fabric_api_key">379f9d1b32592612592b2824a442465e0b8d8c2f</string>
```
separate the fabric key for staging and production

thachnguyendino committed:
- Refactor FragmentExtension
- Create AuthenticationManager to store user.
- Add facebook application id for staging and production.

thachnguyendino committed:
Merge branch 'master_up'

# Conflicts:
#	SportBook/app/src/main/java/com/dinosys/sportbook/di/authentication/AuthenticationComponent.kt
#	SportBook/app/src/main/java/com/dinosys/sportbook/networks/authentication/AuthenticationAPI.kt
#	SportBook/app/src/main/res/values/strings.xml
